validations/src/validations.rs

```diff
@@ -1,17 +1,22 @@
-use witnet_crypto::{hash::Sha256, merkle::merkle_tree_root as crypto_merkle_tree_root};
+use witnet_crypto::{
+    hash::Sha256,
+    merkle::{merkle_tree_root as crypto_merkle_tree_root, ProgressiveMerkleTree},
+    signature::verify,
+};
 
 use std::collections::HashMap;
 use witnet_data_structures::{
     chain::{
-        Block, BlockError, BlockInChain, CheckpointBeacon, DataRequestOutput, Epoch, Hash,
-        Hashable, Input, KeyedSignature, Output, OutputPointer, RADRequest, Signature, Transaction,
-        TransactionBody, TransactionError, TransactionType, TransactionsPool, UnspentOutputsPool,
+        Block, BlockError, BlockInChain, CheckpointBeacon, Epoch, Hash, Hashable, Input,
+        KeyedSignature, Output, OutputPointer, RADRequest, Signature, Transaction, TransactionBody,
+        TransactionError, TransactionType, TransactionsPool, UnspentOutputsPool,
     },
     data_request::DataRequestPool,
     serializers::decoders::{TryFrom, TryInto},
 };
 
-use witnet_crypto::signature::verify;
+use log::debug;
```

Comment:
_Suggestion_: import just log `use log` and then use `log::debug(...)` etc. This way with a grep `log::` you can find all the places where we log.